### PR TITLE
[#152869] Log resolve disputed order

### PR DESCRIPTION
**Pull request by cardoce**
# Release Notes

Log the following event: Who/when someone disputes and resolve an order

### CODE REVIEW

File: app/models/order_detail.rb [hunk 1]
```diff
@@ -890,6 +890,15 @@ def notify_purchaser_of_order_status
     end
   end
 
+  def self.where_order_number_and_id(query)
+    return nil unless /\A(?<order_id>\d+)-(?<id>\d+)\z/ =~ query
+    where(id: id, order_id: order_id)
```
Comment (jhanggi) [inline app/models/order_detail.rb:895]: I wonder if we could maybe now use this in the `GlobalSearch::OrderSearcher`. It's fine for now, though.

File: app/models/order_detail.rb [hunk 2]
```diff
@@ -890,6 +890,15 @@ def notify_purchaser_of_order_status
...
+  end
+
+  def to_log_s
+    "Order # #{order_id}-#{id}"
```
Comment (jhanggi) [inline app/models/order_detail.rb:899]: Can we use `OrderDetail#order_number`? And this is kind of picky, but everywhere we display an order number there's no space between the `#` and the number.

File: app/services/order_details/dispute_resolved_notifier.rb [hunk 3]
```diff
@@ -4,12 +4,20 @@ module OrderDetails
 
   class DisputeResolvedNotifier < SimpleDelegator
 
+    attr_accessor :current_user
+
+    def initialize(order_detail, current_user='')
```
Comment (jhanggi) [inline app/services/order_details/dispute_resolved_notifier.rb:9]: I think this is a good place for a keyword arg. And also, using a a blank string as the default seems a little weird here. Maybe just a nil.

```suggestion
    def initialize(order_detail, current_user: nil)
```

File: config/locales/views/admin/en.log_events.yml [hunk 4]
```diff
@@ -13,7 +13,10 @@ en:
         journal:
           create: Journal created
         statement:
-          create: Statement created
+          create: Statement added
```
Comment (jhanggi) [inline config/locales/views/admin/en.log_events.yml:16]: Why did you change this? I don't have a strong preference for "created" vs "added", but I think we should be consistent with journals.

File: app/models/order_detail.rb (hunk 2 above)
Comment (cardoce) [reply app/models/order_detail.rb:899]: Sure

File: config/locales/views/admin/en.log_events.yml (hunk 4 above)
Comment (cardoce) [reply config/locales/views/admin/en.log_events.yml:16]: Sorry I'll change this back, probably I miss it when I merge master and fixed conflicts.

File: app/services/order_details/param_updater.rb [hunk 5]
```diff
@@ -79,7 +79,7 @@ def update_attributes(params)
   end
 
   def trigger_notifications
-    OrderDetails::DisputeResolvedNotifier.new(@order_detail).notify
+    OrderDetails::DisputeResolvedNotifier.new(@order_detail, @editing_user).notify
```
Comment (jhanggi) [inline app/services/order_details/param_updater.rb:82]: Looks like we need to update this to use the keyword arg now

File: config/locales/views/admin/en.log_events.yml [hunk 6]
```diff
@@ -14,6 +14,9 @@ en:
           create: Journal created
         statement:
           create: Statement created
+        order_detail:
+          dispute: Order disputed
+          resolve: Order resolved
```
Comment (jhanggi) [inline config/locales/views/admin/en.log_events.yml:19]: Let's say `Order dispute resolved` to be a little more specific.
